common/src/main/cpp/extensions/openxr_fb_hand_tracking_aim_extension_wrapper.cpp

```diff
@@ -0,0 +1,188 @@
+/*                                                                        */
+/* The above copyright notice and this permission notice shall be         */
+/* included in all copies or substantial portions of the Software.        */
+/*                                                                        */
+/* THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND,        */
+/* EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF     */
+/* MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. */
+/* IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY   */
+/* CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT,   */
+/* TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE      */
+/* SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.                 */
+/**************************************************************************/
+
+#include "extensions/openxr_fb_hand_tracking_aim_extension_wrapper.h"
+
+#include <godot_cpp/classes/project_settings.hpp>
+#include <godot_cpp/classes/xr_pose.hpp>
+
+using namespace godot;
+
+OpenXRFbHandTrackingAimExtensionWrapper *OpenXRFbHandTrackingAimExtensionWrapper::singleton = nullptr;
+
+OpenXRFbHandTrackingAimExtensionWrapper *OpenXRFbHandTrackingAimExtensionWrapper::get_singleton() {
+	if (singleton == nullptr) {
+		singleton = memnew(OpenXRFbHandTrackingAimExtensionWrapper());
+	}
+	return singleton;
+}
+
+OpenXRFbHandTrackingAimExtensionWrapper::OpenXRFbHandTrackingAimExtensionWrapper() :
+		OpenXRExtensionWrapperExtension() {
+	ERR_FAIL_COND_MSG(singleton != nullptr, "An OpenXRFbHandTrackingAimExtensionWrapper singleton already exists.");
+
+	request_extensions[XR_FB_HAND_TRACKING_AIM_EXTENSION_NAME] = &fb_hand_tracking_aim_ext;
+	singleton = this;
+}
+
+OpenXRFbHandTrackingAimExtensionWrapper::~OpenXRFbHandTrackingAimExtensionWrapper() {
+	cleanup();
+}
+
+void OpenXRFbHandTrackingAimExtensionWrapper::_bind_methods() {
+	ClassDB::bind_method(D_METHOD("is_enabled"), &OpenXRFbHandTrackingAimExtensionWrapper::is_enabled);
+}
+
+void OpenXRFbHandTrackingAimExtensionWrapper::cleanup() {
+	XRServer *xr_server = XRServer::get_singleton();
+
+	for (int i = 0; i < Hand::HAND_MAX; i++) {
+		if (xr_server != nullptr) {
+			xr_server->remove_tracker(trackers[i]);
+		}
+		trackers[i].unref();
+	}
+
+	fb_hand_tracking_aim_ext = false;
+}
+
+godot::Dictionary OpenXRFbHandTrackingAimExtensionWrapper::_get_requested_extensions() {
+	godot::Dictionary result;
+	for (auto ext : request_extensions) {
+		godot::String key = ext.first;
+		uint64_t value = reinterpret_cast<uint64_t>(ext.second);
+		result[key] = (godot::Variant)value;
+	}
+	return result;
+}
+
+PackedStringArray OpenXRFbHandTrackingAimExtensionWrapper::_get_suggested_tracker_names() {
+	PackedStringArray arr = PackedStringArray();
+	arr.push_back("/user/fbhandaim/left");
+	arr.push_back("/user/fbhandaim/right");
```

Comment:
Can you turn these two into constants since they're being reused below as well.

Reply:
Done 👍